consensus/hotstuff/votecollector/combined_vote_processor_v2.go

```diff
@@ -51,8 +52,31 @@ func (f *combinedVoteProcessorFactoryBaseV2) Create(block *model.Block) (hotstuf
 		return nil, fmt.Errorf("could not create aggregator for staking signatures: %w", err)
 	}
 
-	rbRector := &signature.RandomBeaconReconstructor{} // TODO: initialize properly when ready
+	publicKeyShares := make([]crypto.PublicKey, 0, len(allParticipants))
+	dkg, err := f.committee.DKG(block.BlockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get DKG info at block %v: %w", block.BlockID, err)
+	}
+	for _, participant := range allParticipants {
+		pk, err := dkg.KeyShare(participant.NodeID)
+		if err != nil {
```

Comment:
@tarakby Not sure how to behave in this situation, need your input here

Reply:
At this stage, all participants have a public key share that should be returned by `dkg.KeyShare`, regardless of the local DKG results of each participant.